File: sol/1590/1590.py
class Solution:
    def minSubarray(self, nums: List[int], p: int) -> int:
        n = len(nums)
        remainder = sum(nums) % p
        if remainder == 0:
            return 0
        d = {0: -1}
        tot = 0
        res = n
        for i, x in enumerate(nums):
            tot = (tot + x) % p
            j = d.get((tot - remainder + p) % p, -2)
            if j != -2:
                res = min(res, i - j)
            d[tot] = i
        return res if res != n else -1

File: sol/1590/1590.cpp
#include <unordered_map>

struct Solution {
    int minSubarray(vector<int>& nums, int p) {
        int n = nums.size();
        int remainder = 0;
        for (int x : nums){
            remainder = (remainder + x) % p;
        }
        if (remainder == 0){
            return 0;
        }
        unordered_map<int, int> d;
        d[0] = -1;
        int tot = 0, res = n, x;
        for (int i = 0; i < n; ++i){
            tot = (tot + nums[i]) % p;
            x = (tot - remainder + p) % p;
            if (d.contains(x)){
                res = min(res, i - d[x]);
            }
            d[tot] = i;
        }
        return res == n ? -1 : res;
    }
};
